File: 0384-Shuffle an Array/0384-Shuffle an Array.cpp
class Solution {
public:
    Solution(vector<int> nums) {
        this->nums = nums;
    }
    
    /** Resets the array to its original configuration and return it. */
    vector<int> reset() {
        return nums;
    }
    
    /** Returns a random shuffling of the array. */
    vector<int> shuffle() {
        vector<int> seed(nums);
        for (int i = 0; i < nums.size(); ++i)
        {
            int j = rand() % (i + 1);
            swap(seed[i], seed[j]);
        }
        
        return seed;
    }
    
private:
    vector<int> nums;
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * vector<int> param_1 = obj.reset();
 * vector<int> param_2 = obj.shuffle();
 */
 

File: 0301-0400/0384-Shuffle an Array/0384-Shuffle an Array.py
import random
class Solution:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def reset(self) -> List[int]:
        """
        Resets the array to its original configuration and return it.
        """
        return self.nums

    def shuffle(self) -> List[int]:
        """
        Returns a random shuffling of the array.
        """
        result = self.nums[:]
        for i in range(1, len(result)):
            j = random.randrange(i + 1)
            result[i], result[j] = result[j], result[i]
        return result


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.reset()
# param_2 = obj.shuffle()
